tests/run-make/cdylib-dylib-linkage/rmake.rs

```diff
@@ -0,0 +1,39 @@
+// Previously, rustc mandated that cdylibs could only link against rlibs as dependencies,
+// making linkage between cdylibs and dylibs impossible. After this was changed in #68448,
+// this test attempts to link both `foo` (a cdylib) and `bar` (a dylib) and checks that
+// both compilation and execution are successful.
+// See https://github.com/rust-lang/rust/pull/68448
+
+//@ ignore-cross-compile
+// Reason: the compiled binary is executed
+
+use run_make_support::{
+    bin_name, cc, has_extension, has_prefix, has_suffix, is_msvc, name_contains, path, run, rustc,
+    shallow_find_files, target,
+};
+
+fn main() {
+    rustc().arg("-Cprefer-dynamic").input("bar.rs").run();
+    rustc().input("foo.rs").run();
+    let sysroot = rustc().print("sysroot").run().stdout_utf8();
+    let target_sysroot = path(sysroot).join("lib/rustlib").join(target()).join("lib");
```

Comment:
Problem: yes, `--print sysroot` produces a path

```
/path/to/sysroot\n
```

notice the `\n`, you need to trim that off before you join 😄

Reply:
Wow, thank you, I can hardly believe it was that easy. I'm pretty sure I ran into this exact same error a while back, and that you gave me the same advice.